05_stringer/starfallz/main.go

```diff
@@ -0,0 +1,27 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, IPAddr{127, 0, 0, 1})
+}
+
+type IPAddr struct {
+	add1, add2, add3, add4 uint8
+}
+
+func (ipAddr IPAddr) String() string {
```

Comment:
`(ipAddr IPAddr)` This is called receiver in Golang, and a code-owner pointed out to me that the receiver name should be very short, usually one or two letters. The main reason is that the variable will be used straight-away, and won't be confused with other variables in the block. I did my own research as well, and it appears that this is the case. But this might be debatable and not everyone likes this convention.

Reply:
This one is an odd one too. I guess I came from java and usually I am all in on readability and I hate to have comments in my code. Therefore generally I tend to stay away from shortform in naming variables, method, etc. (I would have request changes on the pr if it's Java or even js). But if it's the proper coding pattern of golang then I will have to change my mindset and follow the convention. It would be good if there's a pattern doc that we can follow especially when building enterprise app.

Reply:
This was an odd one for me too, I always believe that variable names should be clear and concise. Guess this is something we need to get use to.

Reply:
Naming convention in go is to use really short names for receivers. The reason is that a method can only have one receiver, and that name should be reused for all methods on a particular type. I don't think having a long receiver name aides readability, I actually think the ridiculously long variable names in java hinders readability.
Also, consider that the `this` keyword is often used to refer to the class instance in a method (or it is omitted entirely). Think of the receiver name as `this`.

Reply:
alright will change it thanks, need to get use to it =)

Reply:
How about you do it then? :)